packages/wrangler/src/r2/object.ts

```diff
@@ -259,9 +259,22 @@ export const r2ObjectPutCommand = createCommand({
 		let object: ReadableStream;
 		let objectSize: number;
 		if (file) {
-			object = await createFileReadableStream(file);
-			const stats = fs.statSync(file);
-			objectSize = stats.size;
+			try {
+				const stats = fs.statSync(file);
+				objectSize = stats.size;
+				object = await createFileReadableStream(file);
+			} catch (err) {
+				if ((err as NodeJS.ErrnoException).code === "ENOENT") {
+					throw new FatalError(`The file "${file}" does not exist.`);
+				}
+				const error = new FatalError(
```

Comment:
FatalError is the same as UserError but just allows you to set a custom process exit code. So in this case you might as set UserError.

I have seen this a few times. I wonder if there is any actual value in having two distinct classes?